app/models/reservation.rb

```diff
@@ -476,7 +477,40 @@ def name
 
   def update_overdue
     return true unless checked_out?
-    self.overdue = due_date < Time.zone.today
+    if due_date < Time.zone.today
+      self.overdue = true
+    else
+      # decrement the counter cache if we're changing overdue
+      # check for id to make sure this is not a new, unsaved reservation
+      # (only relevant in testing when overdue reservations are created)
+      equipment_model.decrement(:overdue_count) if overdue && id
```

Comment:
Yeah, those are the current conditions on the counter cache. 

I think adding `checked_out?` to the condition on decrementing here would solve that problem?